corehq/apps/api/odata/utils.py

```diff
@@ -23,3 +24,35 @@ def get_case_type_to_properties(domain):
             cleaned_case_property = export_group_schema.label.replace('_', '')
             case_type_to_properties[case_type].append(cleaned_case_property)
     return dict(case_type_to_properties)
+
+
+def get_xmlns_to_properties(domain, app_id):
+    return {
+        xmlns: get_properties_by_xmlns(domain, app_id, xmlns)
+        for xmlns in get_xmlns_by_app(domain, app_id)
+    }
+
+
+def get_xmlns_by_app(domain, app_id):
+    app = get_current_app(domain, app_id)
+    return [form.xmlns.split('/')[-1] for form in app.get_forms()]
+
+
+def get_properties_by_xmlns(domain, app_id, xmlns):
+    complete_xmlns = 'http://openrosa.org/formdesigner/' + xmlns
+    form_export_schema = get_latest_form_export_schema(
+        domain, app_id, complete_xmlns
+    ) or FormExportDataSchema.generate_schema_from_builds(domain, app_id, complete_xmlns)
+    export_items = [
+        item for item in form_export_schema.group_schemas[0].items
+        if isinstance(item, ExportItem)
+    ]
+    return set([get_odata_property_from_export_item(item) for item in export_items]) - {''}
+
+
+def get_odata_property_from_export_item(export_item):
+    return format_odata_property_for_power_bi(export_item.label.split('.')[-1])
```

Comment:
Thanks for adding the tests. Now it's much clearer what the method does. But I think that also revealed that a form of the structure:
```xml
<group1><name>Cory</name></group1>
<group2><name>Nick</name></group2>
<group3><name>Farid</name></group3>
```
Will just club all those properties together, yeah? Wouldn't it be more appropriate to create properties like: `group1_name`, `group2_name`, etc. so that the full set of data within different groups can be sent down?

Reply:
Good point.  I checked what exports do and they just use label as is, so I updated OData to follow that pattern: 7c85eb4ca4f994d006f9d9e87a8601b852a71912 and remove periods since PowerBI doesn't like them: 98e388ad9c3b344d3fe8426c02b36ba39c6974f2